downgrade

```diff
@@ -243,25 +267,82 @@ declare -a installed
 cli() {
   while [[ -n "$1" ]]; do
     case "$1" in
-      --)
-        shift
+      --pacman)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN="$1"
+        else
+          gettext "Missing --pacman argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --pacman-conf)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN_CONF="$1"
+        else
+          gettext "Missing --pacman-conf argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --arch)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ARCH="$1"
+        else
+          gettext "Missing --arch argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --ala-url)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ALA_URL="$1"
+        else
+          gettext "Missing --ala-url argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --ala)
+        DOWNGRADE_FROM_ALA=1
+        DOWNGRADE_FROM_CACHE=0
+        ;;
+      --cached)
+        DOWNGRADE_FROM_ALA=0
+        DOWNGRADE_FROM_CACHE=1
+        ;;
+      --nosudo)
+        DOWNGRADE_NOSUDO=1
+        ;;
+      -h | --help)
+        gettext "$(usage)"
+        exit 1
+        ;;
+      --) ;;
```

Comment:
This branch should (now) `shift`, read `$@` into `terms`, and `break`. That way, in the (super duper unlikely) event of a package name that begins with a hyphen, you can use `--` to be able to pass it as an argument, instead of it being swallowed in `pacman_args`.

Reply:
Yes, that would be a good failsafe in that contingency. To keep the code as similar to the latest version of downgrade (before this PR), I reverted the branch to `shift` and `break` for the `pacman_args` instead of `terms`. I also updated the `usage` script to describe the syntax with `--`.